app/code/community/Twispay/Tpay/Helper/Data.php

```diff
@@ -1,8 +1,1097 @@
+                          , Mage_Sales_Model_Order::STATE_CANCELED
+                          , Mage_Sales_Model_Order::STATE_CANCELED
+                          , Mage::helper('tpay')->__(' Order #%s canceled as recurring profile #%s has been canceled.', $recurringProfileChildOrder->getIncrementId(), $profile->getId()));
+    } else {
+      Mage::Log(__FUNCTION__ . Mage::helper('tpay')->__(' Subscription cancel failed: Server returned error: %s', $response->message), Zend_Log::ERR , $this->logFileName, /*forceLog*/TRUE);
+      return;
+    }
+  }
+  /************************** Api END **************************/
+
+
+
+  /************************** Notification START **************************/
+  /**
+   * Get the `jsonRequest` parameter (order parameters as JSON and base64 encoded).
+   *
+   * @param orderData: Array containing the order parameters.
+   *
+   * @return string
+   */
+  public function getBase64JsonRequest(array $orderData) {
+    return base64_encode(json_encode($orderData));
+  }
+
+
+  /**
+   * Get the `checksum` parameter (the checksum computed over the `jsonRequest` and base64 encoded).
+   *
+   * @param orderData: Array containing the order parameters.
+   * @param secretKey: The secret key (from Twispay).
+   *
+   * @return string
+   */
+  public function getBase64Checksum(array $orderData, $secretKey) {
+    $hmacSha512 = hash_hmac(/*algo*/'sha512', json_encode($orderData), $secretKey, /*raw_output*/true);
+    return base64_encode($hmacSha512);
+  }
+  /************************** Notification END **************************/
+
+
+
+  /************************** Response START **************************/
+  /**
+   * Decrypt the response from Twispay server.
+   *
+   * @param tw_encryptedMessage: - The encripted server message.
+   * @param tw_secretKey:        - The secret key (from Twispay).
+   *
+   * @return Array([key => value,]) - If everything is ok array containing the decrypted data.
+   *         bool(FALSE)            - If decription fails.
+   */
+  public function twispay_tw_decrypt_message($tw_encryptedMessage, $tw_secretKey) {
+    $encrypted = (string)$tw_encryptedMessage;
+
+    if(!strlen($encrypted) || (FALSE == strpos($encrypted, ','))) {
+      return FALSE;
+    }
+
+    /* Get the IV and the encrypted data */
+    $encryptedParts = explode(/*delimiter*/',', $encrypted, /*limit*/2);
+    $iv = base64_decode($encryptedParts[0]);
+    if(FALSE === $iv) {
+      return FALSE;
+    }
+
+    $encryptedData = base64_decode($encryptedParts[1]);
+    if(FALSE === $encryptedData) {
+      return FALSE;
+    }
+
+    /* Decrypt the encrypted data */
+    $decryptedResponse = openssl_decrypt($encryptedData, /*method*/'aes-256-cbc', $tw_secretKey, /*options*/OPENSSL_RAW_DATA, $iv);
+    if(FALSE === $decryptedResponse) {
+      return FALSE;
+    }
+
+    /* JSON decode the decrypted data. */
+    return json_decode($decryptedResponse, /*assoc*/TRUE, /*depth*/4);

```

Comment:
We should also check if the response was decoded successfully with `json_decode`, otherwise return false.

Reply:
Added the check.